### DISCUSSION TITLE
Loading script files with type=text/javascript or some other custom tags?

### DISCUSSION THREAD

**Discussion by oMatej** [vercel/next.js#15821] [Help]
Is there any way to load Next.js chunks (`<NextScript />`) with the `type="text/javascript"` or some other custom tags? 

It seems that there are third-party tools, like `OneTrust` to handle cookie or third-party scripts consent regulation that might prevent loading necessary JS chunks because it might automatically categorize them as some functional / tracking scripts.

**Comment by Oikio** [comment]
There are several ways to conditionally load scripts with Webpack, you can use conditional async `import()` or sync conditional `require()`.
```
if (poorSoulApplied) {
  const yetAnotherAdService = require('./module')
}
```
If it's on the third party CDNs, you can use conditional <script> tags in Head.

**Reply by oMatej** [reply]
Hm, I think I wrote it not clear enough. What I meant was adding custom tags to the scripts loaded by `<NextScript />`. Is there any way to do that? So that instead of for example:
```
<script src="/_next/static/chunks/4998e2149eb89955594d724fcbdc5156fffa2462.b0a780aaaa64f5a45f2d.js" async=""></script>
```
the output would look like
```
<script src="/_next/static/chunks/4998e2149eb89955594d724fcbdc5156fffa2462.b0a780aaaa64f5a45f2d.js" async="" type="text/javascript" data-custom-name="true"></script>
```

**Reply by Oikio** [reply]
Can't help in this case, but I guess it should be configurable in webpack, try to look into it's documentation.

**Comment by sfeir-pach** [comment]
I got same problem I want to include OneTrust cookies consent but sometime is display and other time no... I wonder if the include in header (_document) work as explain in the nextJs doc
